numbat-cli/src/main.rs

```diff
@@ -315,10 +315,21 @@ impl Cli {
         let result = self.repl_loop(&mut rl, interactive);
 
         if interactive {
-            rl.save_history(&history_path).context(format!(
-                "Error while saving history to '{}'",
-                history_path.to_string_lossy()
-            ))?;
+            // Try to save the history, and if it fails because prerequisite directories don't exist, create them.
+            if let Err(e) = rl.save_history(&history_path) {
```

Comment:
Instead of checking whether it fails, we could always call `create_dir_all`.

Reply:
Let me know if the latest changes are more or less what you had in mind.

Reply:
Let me know if the latest changes are more or less what you had in mind.